persistence-jpa/javadsl/src/main/java/com/lightbend/lagom/javadsl/persistence/jpa/JpaSession.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.javadsl.persistence.jpa;
+
+import javax.persistence.EntityManager;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+/**
+ * Provides asynchronous access to a JPA EntityManager. Inject this into classes that require access to JPA, including
+ * JPA read-side implementations.
+ *
+ * @since 1.3
+ */
+public interface JpaSession {
+    /**
+     * Execute the given function in a JPA transaction.
+     * <p>
+     * The JPA EntityManager is provided to the function. This will execute the callback in a thread pool that is
+     * specifically designed for use with JDBC calls.
+     *
+     * @param block the function to execute
+     * @return a completion stage that will complete with the result of the supplied block
+     */
+    <T> CompletionStage<T> withTransaction(Function<EntityManager, T> block);
```

Comment:
I ended up taking a different approach with this than previously discussed: just providing a single `withTransaction` method that takes a `Function`.

I couldn't think of a use for the previous `withEntityManager` method that doesn't include transaction management, so I'll wait for that to be requested before exposing it.

I also removed the version of `withTransaction` that took a `Consumer<EntityManager>`, as this can be simulated with a `Function<EntityManager, Done>`. Functions returning `Done` are already a common pattern in the Lagom code base.

As a convenience, it might be nice for us to provide a utility function that wraps a `Consumer<T>` into a `Function<T, Done>`. This is something that I would have found useful in a number of places (and ended up implementing in the online auction app in [`CompletionStageUtils`](https://github.com/lagom/online-auction-java/blob/master/item-impl/src/main/java/com/example/auction/item/impl/CompletionStageUtils.java#L30-L35).

Any thoughts on what might be a good place to put it in Lagom, or what a good name for the function would be (I'm not really happy with `accept`, as it is currently called in online auction).